### PR TITLE
[onert] Add shape inference for squeeze and its testcases

### PR DESCRIPTION
**Pull request by KimDongEon**
This pr enables shape inference for squeeze

Signed-off-by: Dongeon kim <good.kim@samsung.com>

### CODE REVIEW

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    const auto &input = _tensor_registry->getITensor(input_idx);
+
+    if (!(input->is_dynamic())) {
+        return;
+    }
+
+    auto input_shape = getShape(input.get());
+
+    // Squeeze output shpae
+    ir::Shape new_shape = inferSqueezeShape(input.info().shape(), op.param());
```
Comment (hasw7569) [inline runtime/onert/core/src/util/shapeinf/Squeeze.cc:99]: you already get shape of input "input_shape = getShape(input.get());"
so I think you don't need to invoke input.info().shape()

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape inferSqueezeShape(const ir::Shape &in_shape, const ir::operation::Squeeze::Param &param)
```
Comment (hasw7569) [inline runtime/onert/core/src/util/shapeinf/Squeeze.cc:24]: could you add prototype of inferSqueezeShape() in Shapeinference.h file ?

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+{
+    const int ndims = param.ndim;
+    const int* squeeze_dims = param.dims;
+    bool should_squeeze[8] = {false};
+    int num_squeezed_dims = 0;
+    if (ndims == 0) {//전체 dims이 1인것 제거
```
Comment (hasw7569) [inline runtime/onert/core/src/util/shapeinf/Squeeze.cc:30]: could you please change annotation of 30 line to english ??

**Comment by dr-venkman** [comment]
Perhaps include positive and (if applicable) negative testcases for squeeze under `test/util/ShapeInference.cc`?

File: tests/nnapi/specs/V1_1/squeeze_dynamic_float_nnfw.mod.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+input0_shape = [4, 1]
+
+dynamic_layer = dynamic_tensor.DynamicInputGenerator(model, input0_shape, "TENSOR_FLOAT32")
+
+input0 = dynamic_layer.getTestNodeInput() # first input of squeeze is dynamic tensor
+
+output = Output("output", "TENSOR_FLOAT32", "{4}")
+
+model = model.Operation("SQUEEZE", input0).To(output)
```
Comment (YiHyunJin) [inline tests/nnapi/specs/V1_1/squeeze_dynamic_float_nnfw.mod.py:30]: You need to add parameters.

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                should_squeeze[idx] = true;
+                ++num_squeezed_dims;
+            }
+        }
+    } else {
+        for (int idx = 0; idx < ndims; ++idx) {
+            int current = squeeze_dims[idx];
+            if (current < 0) {
+                current = squeeze_dims[idx] + in_shape.rank();
+            } else {
```
Comment (YiHyunJin) [inline runtime/onert/core/src/util/shapeinf/Squeeze.cc:43]: Why does it need this?

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                current = squeeze_dims[idx];
+            }
+
+            if (!(current >= 0 && current < in_shape.rank() && in_shape.dim(current) == 1)) {
```
Comment (YiHyunJin) [inline runtime/onert/core/src/util/shapeinf/Squeeze.cc:47]: It use in_shape.rank() frequently, So how about using it as a variable?
`
int rank = in_shape.rank();
`

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+ir::Shape inferSqueezeShape(const ir::Shape &in_shape, const ir::operation::Squeeze::Param &param)
+{
+    const int ndims = param.ndim;
+    const int* squeeze_dims = param.dims;
+    bool should_squeeze[8] = {false};
+    int num_squeezed_dims = 0;
+    if (ndims == 0) {//전체 dims이 1인것 제거
+        for (int idx = 0; idx < in_shape.rank(); ++idx)
+        {
+            if (in_shape.dim(idx) == 1) {
```
[the remark this answers is not in the document]
Comment (KimDongEon) [reply runtime/onert/core/src/util/shapeinf/Squeeze.cc]: I will modify

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc (hunk 5 above)
Comment (KimDongEon) [reply runtime/onert/core/src/util/shapeinf/Squeeze.cc]: I will modify "if (current < 0)" code

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc (hunk 6 above)
Comment (KimDongEon) [reply runtime/onert/core/src/util/shapeinf/Squeeze.cc]: okay
Modify

File: tests/nnapi/specs/V1_1/squeeze_dynamic_float_nnfw.mod.py (hunk 4 above)
Comment (KimDongEon) [reply tests/nnapi/specs/V1_1/squeeze_dynamic_float_nnfw.mod.py]: I will add parameters

**Comment by YiHyunJin** [comment]
@nnfw-bot test all

**Comment by KimDongEon** [comment]
@nnfw-bot test all

**Comment by KimDongEon** [comment]
@hyunsik-yoon
okay
I add WG2 reviewer

**Comment by hyunsik-yoon** [comment]
Can I ask a favor of you?
Once anyone starts reviewing the PR, could you commit additional changes with `commit -m "short_message_without_signoff"`?

That way, reviewers can easily recognized what's been changed from the original commit. Otherwise, reviewers have to read the whole code repeatedly.

(please refer to an example: `Extend it to all for gradlew` commit in the middle of https://github.com/Samsung/ONE/pull/1776)

Thanks!

File: runtime/onert/core/src/util/shapeinf/Squeeze.cc (hunk 2 above)
Comment (KimDongEon) [reply runtime/onert/core/src/util/shapeinf/Squeeze.cc:24]: The prototype was removed again.
It is not necessary because it will only be used on squeeze.

**Pull request merged**